src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +52,123 @@
 public class TranscriptServlet extends HttpServlet {
   private static final String TRANSCRIPT_XML_URL_TEMPLATE =
       "http://video.google.com/timedtext?lang=en&v=";
-  private static final String START_ATTRIBUTE = "start";
-  private static final String DURATION_ATTRIBUTE = "dur";
-  private static final String TEXT_TAG = "text";
-  private static final String TEST_VIDEO_ID = "3ymwOvzhwHs";
+  private static final String ATTR_START = "start";
+  private static final String ATTR_DURATION = "dur";
+  private static final String TAG_TEXT = "text";
+  private static final String PARAM_LECTURE = "lecture";
+  private static final String PARAM_LECTURE_ID = "id";
+  private static final String PARAM_VIDEO_ID = "video";
+
+  private static DatastoreService datastore;
 
   @Override
   public void init() throws ServletException {
-    // TODO: Implement Transcript.
+    datastore = DatastoreServiceFactory.getDatastoreService();
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: Pass the video ID from another servlet.
-    String transcriptXMLUrl = TRANSCRIPT_XML_URL_TEMPLATE + TEST_VIDEO_ID;
-    DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
-    final DocumentBuilder documentBuilder;
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Document document = getTranscriptXmlAsDocument(request).get();
+    putTranscriptLinesInDatastore(request, document);
+  }
+
+  /**
+   * Gets the transcript for a video as a document. Otherwise, return Optional.empty()
+   * if there is a parsing error. {@code request} indicates the video to extract
+   * the transcript from.
+   */
+  private Optional<Document> getTranscriptXmlAsDocument(HttpServletRequest request)
+      throws IOException {
+    String videoId = request.getParameter(PARAM_VIDEO_ID);
+    String transcriptXMLUrl = TRANSCRIPT_XML_URL_TEMPLATE + videoId;
+
     final Document document;
     try {
-      documentBuilder = documentBuilderFactory.newDocumentBuilder();
+      DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
       document = documentBuilder.parse(new URL(transcriptXMLUrl).openStream());
       document.getDocumentElement().normalize();
     } catch (ParserConfigurationException | SAXException e) {
       // TODO: Alert the user.
       System.out.println("XML parsing error");
-      return;
+      return Optional.empty();
     }
-    NodeList nodeList = document.getElementsByTagName(TEXT_TAG);
+    return Optional.of(document);
+  }
+
+  /**
+   * Puts each transcript line from {@code document} in datastore as its own entity. The lecture
+   * key to group the transcript lines under is indicated in {@code request}.
```

Comment:
"The lecture key to group..." Is an implementation detail. Can you rephrase it?

Reply:
I made it a @param.

Done!